[DOC-9753] Document minimum supported vCPU per node for Dedicated on Azure (#18320)
cockroachdb/docs: src/current/cockroachcloud/production-checklist.md

`raw data (storage, in GB) * replication factor (3 by default) * remove 40% to account for compression (0.6) * headroom (1.5-2)`

{{site.data.alerts.callout_info}}

> CockroachDB Dedicated clusters can be created with a minimum of 2vCPUs per node on AWS and GCP or 4vcPUs per node on Azure.

Reviewer: Should we use the product variable for Dedicated?

> CockroachDB Dedicated clusters can be created with a minimum of 2vCPUs per node on AWS and GCP or 4vcPUs per node on Azure.

Author: ```suggestion
CockroachDB {{ site.data.products.dedicated }} clusters can be created with a minimum of 2vCPUs per node on AWS and GCP or 4vcPUs per node on Azure.
```

> CockroachDB Dedicated clusters can be created with a minimum of 2vCPUs per node on AWS and GCP or 4vcPUs per node on Azure (Limited Access).

Author: ```suggestion
CockroachDB Dedicated clusters can be created with a minimum of 2vCPUs per node on AWS and GCP or 4vcPUs per node on Azure.
```

> CockroachDB {{ site.data.products.dedicated }} clusters can be created with a minimum of 2vCPUs per node on AWS and GCP or 4vcPUs per node on Azure.

Author: ```suggestion
CockroachDB {{ site.data.products.dedicated }} clusters can be created with a minimum of 2 vCPUs per node on AWS and GCP or 4 vCPUs per node on Azure.
```
